src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.CreateOrderRequestDto;
+import com.example.onlinebookstore.dto.OrderDto;
+import com.example.onlinebookstore.dto.OrderItemDto;
+import com.example.onlinebookstore.dto.UpdateStatusOrderDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.service.OrderService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import com.example.onlinebookstore.service.UserService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final UserService userService;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
```

Comment:
```suggestion
    @Override
    @Transactional
```
Please don't forget about this annotation, it is very important (think in which else methods it is needed too)